gcp/biganimal_GCP_baserole.yaml

```diff
@@ -0,0 +1,540 @@
+- container.storageClasses.create
+- container.storageClasses.delete
+- container.storageClasses.get
+- container.storageClasses.list
+- container.storageClasses.update
+- container.subjectAccessReviews.create
+- container.thirdPartyObjects.create
+- container.thirdPartyObjects.delete
+- container.thirdPartyObjects.get
+- container.thirdPartyObjects.list
+- container.thirdPartyObjects.update
+- container.tokenReviews.create
+- container.updateInfos.create
+- container.updateInfos.delete
+- container.updateInfos.get
+- container.updateInfos.list
+- container.updateInfos.update
+- container.validatingWebhookConfigurations.create
+- container.validatingWebhookConfigurations.delete
+- container.validatingWebhookConfigurations.get
+- container.validatingWebhookConfigurations.list
+- container.validatingWebhookConfigurations.update
+- container.volumeAttachments.create
+- container.volumeAttachments.delete
+- container.volumeAttachments.get
+- container.volumeAttachments.getStatus
+- container.volumeAttachments.list
+- container.volumeAttachments.update
+- container.volumeAttachments.updateStatus
+- container.volumeSnapshotClasses.create
+- container.volumeSnapshotClasses.delete
+- container.volumeSnapshotClasses.get
+- container.volumeSnapshotClasses.list
+- container.volumeSnapshotClasses.update
+- container.volumeSnapshotContents.create
+- container.volumeSnapshotContents.delete
+- container.volumeSnapshotContents.get
+- container.volumeSnapshotContents.getStatus
+- container.volumeSnapshotContents.list
+- container.volumeSnapshotContents.update
+- container.volumeSnapshotContents.updateStatus
+- container.volumeSnapshots.create
+- container.volumeSnapshots.delete
+- container.volumeSnapshots.get
+- container.volumeSnapshots.getStatus
+- container.volumeSnapshots.list
+- container.volumeSnapshots.update
+- container.volumeSnapshots.updateStatus
+- dns.changes.create
+- dns.changes.get
+- dns.changes.list
+- dns.gkeClusters.bindDNSResponsePolicy
+- dns.gkeClusters.bindPrivateDNSZone
+- dns.managedZones.create
+- dns.managedZones.delete
+- dns.managedZones.get
+- dns.managedZones.getIamPolicy
+- dns.managedZones.list
+- dns.managedZones.update
+- dns.networks.bindDNSResponsePolicy
+- dns.networks.bindPrivateDNSPolicy
+- dns.networks.bindPrivateDNSZone
+- dns.networks.targetWithPeeringZone
+- dns.networks.useHealthSignals
+- dns.projects.get
+- dns.resourceRecordSets.create
+- dns.resourceRecordSets.delete
+- dns.resourceRecordSets.get
+- dns.resourceRecordSets.list
+- dns.resourceRecordSets.update
+- iam.roles.create
+- iam.roles.delete
+- iam.roles.get
+- iam.roles.list
+- iam.roles.update
+- iam.serviceAccounts.actAs
+- iam.serviceAccounts.create
+- iam.serviceAccounts.delete
+- iam.serviceAccounts.disable
+- iam.serviceAccounts.enable
+- iam.serviceAccounts.get
+- iam.serviceAccounts.getIamPolicy
+- iam.serviceAccounts.list
+- iam.serviceAccounts.setIamPolicy
+- iam.serviceAccounts.undelete
+- iam.serviceAccounts.update
```

Comment:
I think there's general awareness that this set is problematic, but also that this needs dedicated work to be improved across the CSP board (here we're staying consistent with AZ, conceptually). Worth putting a comment just so it's official.

We know we'll have to look into AWS boundary equivalent mechanisms, narrowing permissions scopes, and in general ensure as few identities as possible have roles that allow to create/edit roles and "role bindings" (policies here, I guess).

I assume an identity that implicitly an identity that can create/edit roles or bind roles to identities can't create roles with permissions it doesn't itself have, or bind permissions it doesn't itself have. If not the point above becomes even more relevant.

Reply:
I agree. On GCP if I'm right (don't know so well GCP as I do for AWS/Azure, so feel free to correct me if I'm wrong) there is no way to restrict the roles creation/updation/deletion with a condition that restricts this to be done only on a subset of roles for example based on names like we currently do for AWS [here](https://github.com/EnterpriseDB/cloud-utilities/blob/main/aws/biganimal_AWS_addonpolicy_v2.json#L64-L71)... Also, for dynamic service account/roles creations/updations (for example for Workload Identities that belongs to Postgres clusters or Kubernetes operators) we could try to investigate how to restrict the rights that the main "BigAnimal Role" has setting roles on dynamically created service accounts with [Limit role granting](https://cloud.google.com/iam/docs/setting-limits-on-granting-roles#limiting-role-granting), but I suppose that this list of "allowed roles" also can't contains wildcards or "non existing roles" at the time of the main "BigAnimal Role" binding, so I don't know in detail now if that path is feasible or not...